gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleProjectProvider.java

```diff
@@ -61,4 +79,71 @@ public synchronized ILiferayProject provide( Object adaptable )
         return retval;
     }
 
+
+    @Override
+    public IStatus createNewProject( Object operation, IProgressMonitor monitor ) throws CoreException
+    {
+        IStatus retval = null;
+
+        if( ! (operation instanceof NewLiferayModuleProjectOp ) )
+        {
+            throw new IllegalArgumentException( "Operation must be of type NewLiferayModuleProjectOp" ); //$NON-NLS-1$
+        }
+
+        final NewLiferayModuleProjectOp op = NewLiferayModuleProjectOp.class.cast( operation );
+
+        final String projectName = op.getProjectName().content();
+
+        IPath location = PathBridge.create( op.getLocation().content() );
+
+        // for location we should use the parent location
+        if( location.lastSegment().equals( projectName ) )
+        {
+            // use parent dir since maven archetype will generate new dir under this location
+            location = location.removeLastSegments( 1 );
+        }
+
+        final String projectType = op.getProjectTemplate().content().toString();
+
+        retval =
+            createOSGIBundleProject(
+                location.toFile(), location.toFile(), projectType, ProjectBuild.gradle.toString(), projectName,
+                projectName, projectName );
+
+        if( retval.isOK() )
+        {
+            ProjectImportConfiguration configuration = new ProjectImportConfiguration();
+            GradleDistributionWrapper from = GradleDistributionWrapper.from( GradleDistribution.fromBuild() );
+            configuration.setGradleDistribution( from );
+            configuration.setProjectDir( location.append( projectName ).toFile() );
+            configuration.setApplyWorkingSets( false );
+            configuration.setWorkingSets( new ArrayList<String>() );
+            new SynchronizeGradleProjectJob(
+                configuration.toFixedAttributes(), configuration.getWorkingSets().getValue(),
+                AsyncHandler.NO_OP ).schedule();
+        }
+
+        return retval;
+    }
+
+    @Override
+    public IStatus validateProjectLocation( String projectName, IPath path )
+    {
+        IStatus retval = Status.OK_STATUS;
+        return retval;
+    }
+
+    @Override
+    public IStatus createOSGIBundleProject(
```

Comment:
Both maven adn gradle projects just call through right to ProjectUtil.createOSGIBundlePRoject().  this makes the interface not seem worth it if there is no difference at all in the implementation.  Why the need for subclasses?